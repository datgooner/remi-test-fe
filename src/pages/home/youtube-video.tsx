import { AspectRatio } from "@/components/ui/aspect-ratio";
import { Card } from "@/components/ui/card";
import { Label } from "@/components/ui/label";
import { ScrollArea } from "@/components/ui/scroll-area";
import { VideoModel } from "@/model/video.model";
import { memo } from "react";

interface YoutubeVideoProps extends VideoModel {}
const YoutubeVideo = memo(
  ({ embedUrl, title, description, createBy }: YoutubeVideoProps) => {
    return (
      <Card className="flex w-full flex-col space-y-8 p-4 md:flex-row md:space-x-8 md:space-y-0">
        <div className="flex-1">
          <AspectRatio ratio={16 / 9}>
            <iframe
              src={embedUrl}
              title={title}
              frameBorder="0"
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
              referrerPolicy="strict-origin-when-cross-origin"
              allowFullScreen
              className="h-full w-full"
              loading="lazy"
            />
          </AspectRatio>
        </div>

        <div className="prose flex max-h-[50vh] flex-col md:w-[clamp(16rem,25vw,70rem)]">
          <h2>{title}</h2>
          <div>
            <Label className="text-base">Share by: </Label>
            <span className="font-semibold">{createBy.email}</span>
          </div>
          <Label className="mb-2 text-base">Description:</Label>
          <ScrollArea className="flex-1">
            <p className="mt-0 whitespace-pre-wrap break-words text-sm">
              {description}
            </p>
          </ScrollArea>
        </div>
      </Card>
    );
  }
);

export { YoutubeVideo };
