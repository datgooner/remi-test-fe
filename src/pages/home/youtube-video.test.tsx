import { render } from "@testing-library/react";
import { YoutubeVideo, YoutubeVideoSkeleton } from "./youtube-video";

const mockVideo = {
  embedUrl: "https://www.youtube.com/embed/123456",
  title: "Test Video",
  description: "This is a test video description.",
  createBy: { email: "test@example.com", _id: "userId" },
  _id: "id",
  url: "url",
  videoId: "videoId",
  createdAt: "",
  updatedAt: "",
};

describe("YoutubeVideo component", () => {
  it("to match snapshot", () => {
    const tree = render(<YoutubeVideo {...mockVideo} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders youtube video details correctly", () => {
    const { getByText, getByTestId } = render(<YoutubeVideo {...mockVideo} />);

    // Check if title, email, and description are rendered
    expect(getByText(mockVideo.title)).toBeInTheDocument();
    expect(getByText(`${mockVideo.createBy.email}`)).toBeInTheDocument();
    expect(getByText(mockVideo.description)).toBeInTheDocument();

    // Check if the iframe is rendered with correct attributes
    const iframe = getByTestId("video-iframe");
    expect(iframe).toBeInTheDocument();
    expect(iframe.getAttribute("src")).toBe(mockVideo.embedUrl);
    expect(iframe.getAttribute("title")).toBe(mockVideo.title);
    expect(iframe.getAttribute("frameBorder")).toBe("0");
    expect(iframe.getAttribute("allow")).toBe(
      "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    );
    expect(iframe.getAttribute("referrerPolicy")).toBe(
      "strict-origin-when-cross-origin"
    );
    expect(iframe.getAttribute("allowFullScreen")).toBe("");
    expect(iframe.getAttribute("class")).toContain("h-full w-full");
    expect(iframe.getAttribute("loading")).toBe("lazy");
  });
});

describe("YoutubeVideoSkeleton component", () => {
  it("to match snapshot", () => {
    const tree = render(<YoutubeVideoSkeleton />);
    expect(tree).toMatchSnapshot();
  });
});
