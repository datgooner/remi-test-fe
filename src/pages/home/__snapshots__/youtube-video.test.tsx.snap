// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YoutubeVideo component > to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow flex w-full flex-col space-y-8 p-4 md:flex-row md:space-x-8 md:space-y-0"
      >
        <div
          class="flex-1"
        >
          <div
            data-radix-aspect-ratio-wrapper=""
            style="position: relative; width: 100%; padding-bottom: 56.25%;"
          >
            <div
              style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
            >
              <iframe
                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
                allowfullscreen=""
                class="h-full w-full"
                data-testid="video-iframe"
                frameborder="0"
                loading="lazy"
                referrerpolicy="strict-origin-when-cross-origin"
                src="https://www.youtube.com/embed/123456"
                title="Test Video"
              />
            </div>
          </div>
        </div>
        <div
          class="prose flex max-h-[50vh] flex-col md:w-[clamp(16rem,25vw,70rem)]"
        >
          <h2>
            Test Video
          </h2>
          <div>
            <label
              class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
            >
              Share by: 
            </label>
            <span
              class="font-semibold"
            >
              test@example.com
            </span>
          </div>
          <label
            class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 mb-2 text-base"
          >
            Description:
          </label>
          <div
            class="relative overflow-hidden flex-1"
            dir="ltr"
            style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
          >
            <style>
              [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
            </style>
            <div
              class="h-full w-full rounded-[inherit]"
              data-radix-scroll-area-viewport=""
              style="overflow-x: hidden; overflow-y: scroll;"
            >
              <div
                style="min-width: 100%; display: table;"
              >
                <p
                  class="mt-0 whitespace-pre-wrap break-words text-sm"
                >
                  This is a test video description.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow flex w-full flex-col space-y-8 p-4 md:flex-row md:space-x-8 md:space-y-0"
    >
      <div
        class="flex-1"
      >
        <div
          data-radix-aspect-ratio-wrapper=""
          style="position: relative; width: 100%; padding-bottom: 56.25%;"
        >
          <div
            style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
              allowfullscreen=""
              class="h-full w-full"
              data-testid="video-iframe"
              frameborder="0"
              loading="lazy"
              referrerpolicy="strict-origin-when-cross-origin"
              src="https://www.youtube.com/embed/123456"
              title="Test Video"
            />
          </div>
        </div>
      </div>
      <div
        class="prose flex max-h-[50vh] flex-col md:w-[clamp(16rem,25vw,70rem)]"
      >
        <h2>
          Test Video
        </h2>
        <div>
          <label
            class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
          >
            Share by: 
          </label>
          <span
            class="font-semibold"
          >
            test@example.com
          </span>
        </div>
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 mb-2 text-base"
        >
          Description:
        </label>
        <div
          class="relative overflow-hidden flex-1"
          dir="ltr"
          style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
        >
          <style>
            [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
          </style>
          <div
            class="h-full w-full rounded-[inherit]"
            data-radix-scroll-area-viewport=""
            style="overflow-x: hidden; overflow-y: scroll;"
          >
            <div
              style="min-width: 100%; display: table;"
            >
              <p
                class="mt-0 whitespace-pre-wrap break-words text-sm"
              >
                This is a test video description.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`YoutubeVideoSkeleton component > to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow flex w-full flex-col space-y-8 p-4 md:flex-row md:space-x-8 md:space-y-0"
      >
        <div
          class="flex-1"
        >
          <div
            data-radix-aspect-ratio-wrapper=""
            style="position: relative; width: 100%; padding-bottom: 56.25%;"
          >
            <div
              style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
            >
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton h-full w-full"
                >
                  ‌
                </span>
                <br />
              </span>
            </div>
          </div>
        </div>
        <div
          class="prose flex max-h-[50vh] flex-col md:w-[clamp(16rem,25vw,70rem)]"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton w-10"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow flex w-full flex-col space-y-8 p-4 md:flex-row md:space-x-8 md:space-y-0"
    >
      <div
        class="flex-1"
      >
        <div
          data-radix-aspect-ratio-wrapper=""
          style="position: relative; width: 100%; padding-bottom: 56.25%;"
        >
          <div
            style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton h-full w-full"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </div>
      </div>
      <div
        class="prose flex max-h-[50vh] flex-col md:w-[clamp(16rem,25vw,70rem)]"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton w-10"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
